### PR TITLE
Merge animations before writing to JSON.

### LINKED ISSUE #66: Animations should be grouped by action, not by target node
It seems that COLLADA2GLTF will output one animation entry for each affected node. This causes side effects: https://github.com/KhronosGroup/glTF/issues/1052 and https://github.com/KhronosGroup/glTF-Sample-Models/issues/108#issuecomment-320258251. For example, the Monster model contains 32 animations (1 for each bone) and the viewer must play all of them at once in order to see the "walking" animation properly.

By default, I believe COLLADA2GLTF should merge all of these channels into one animation entry, unless the COLLADA asset explicitly provides multiple actions.

### PR DESCRIPTION
**Pull request by donmccurdy**
Fixes #66.

Related:
- https://github.com/KhronosGroup/glTF-Sample-Models/issues/108
- https://github.com/KhronosGroup/glTF/issues/1052

Disclaimer: I have no idea how memory management works in C++. Do i need to deallocate the old
 animations somehow? 😅

### CODE REVIEW

File: GLTF/src/GLTFAsset.cpp [hunk 1]
```diff
@@ -341,6 +341,24 @@ std::vector<GLTF::BufferView*> GLTF::Asset::getAllCompressedBufferView() {
...
+void GLTF::Asset::mergeAnimations() {
+	if (animations.size() == 0) { return; }
+
+	GLTF::Animation* mergedAnimation = new GLTF::Animation();
+
+	// Merge all animations. In the future, animations should be grouped
+	// according to COLLADA <animation_clip/> nodes.
+	for (size_t i = 0; i < animations.size(); i++) {
+		GLTF::Animation* animation = animations[i];
+		for (GLTF::Animation::Channel* channel : animation->channels) {
```
Comment (najadojo) [inline GLTF/src/GLTFAsset.cpp]: I see this pattern in this file already:

	for (GLTF::Animation* animation : animations) {
		for (GLTF::Animation::Channel* channel : animation->channels) {

**Comment by pjcozzi** [comment]
@lasalvavida are you able to take a quick look at this?

**Comment by lasalvavida** [comment]
This looks good to me, thanks @donmccurdy!

And yeah, don't worry about memory. It's on the roadmap (see #70), but not super high priority since as far as I know people just use this as a command-line tool. 

If there is interest in using the `GLTF` submodule of this project in other applications it will become a higher priority.

**Comment by donmccurdy** [comment]
Thanks @lasalvavida! Is there any scripting to do bulk updates like https://github.com/KhronosGroup/glTF-Sample-Models/pull/76, or would I do that manually?

**Comment by lasalvavida** [comment]
@donmccurdy, check out the tools branch. 

It was never added to master, but I did write a [python script](https://github.com/KhronosGroup/glTF-Sample-Models/blob/tools/tools/generateSampleModels.py) for autogenerating the models.

**Comment by donmccurdy** [comment]
Ok, perfect! 🙂

File: GLTF/src/GLTFAsset.cpp [hunk 2]
```diff
@@ -341,6 +341,23 @@ std::vector<GLTF::BufferView*> GLTF::Asset::getAllCompressedBufferView() {
 	return compressedBufferViews;
 }
 
+void GLTF::Asset::mergeAnimations() {
+	if (animations.size() == 0) { return; }
+
+	GLTF::Animation* mergedAnimation = new GLTF::Animation();
+
+	// Merge all animations. In the future, animations should be grouped
+	// according to COLLADA <animation_clip/> nodes.
```
Comment (emackey) [inline GLTF/src/GLTFAsset.cpp]: Did a new issue get opened to track this?

Comment (emackey) [reply GLTF/src/GLTFAsset.cpp]: I made one.

**Pull request merged**